prov/efa/src/efa_fork_support.c

```diff
@@ -57,8 +57,17 @@ void efa_fork_support_request_initialize()
 	 * libibverbs' fork support are set. These variables are
 	 * defined by ibv_fork_init(3).
 	 */
-	if (fork_support_requested || getenv("RDMAV_FORK_SAFE") || getenv("IBV_FORK_SAFE"))
+	if (fork_support_requested || getenv("RDMAV_FORK_SAFE") || getenv("IBV_FORK_SAFE")) {
 		g_efa_fork_status = EFA_FORK_SUPPORT_ON;
+		/* libibvers's fork support does not work for huge page.
+		 * libibvers does have huge page fork support, which can be activated by
+		 * setting environment RDMAV_HUGEPAGE_SAFE to 1. However, huge page fork
+		 * support is extremely expensive, enabling it would defeat the purpose
+		 * of using huge page. Therefore, we simply disable huge page usage
+		 * when libfabric's fork support is enabled.
+		 */
+		efa_env.use_huge_page = false;
```

Comment:
I'm not sure about this coupling. The logic would be simpler if we always respect the user environment variable even if that could cause issues, e.g. user wants to use huge page and also asks for fork support - we just print a warning.

Accordingly we can advise users to disable hugepage explicitly.

Reply:
I think the point to not overwrite user setting is fair.

I will add additional variable to record that  user set `FI_EFA_HUGE_PAGE`, but we will have to abort() here, because of the risk of silent data corruption.

Reply:
We should add a check here, if FI_EFA_USE_HUGE_PAGE is explicitly set, we should warn

Reply:
> we will have to abort() here, because of the risk of silent data corruption.

Whether data corruption occurs also depends on how the application manages memory, so technically it could/may be ok to proceed.

But I'm ok with abort given the very bad worst case. It also makes testing easier, i.e. catch exception.

Reply:
>Whether data corruption occurs also depends on how the application manages memory, so technically it could/may be ok to proceed.

the corruption if happens will be silent, so user might not know it happened.

Reply:
added the abort in new revision, also added a test to verify